kubernetes/base/watch/watch.py

```diff
@@ -172,14 +172,19 @@ def stream(self, func, *args, **kwargs):
         # We want to ensure we are returning within that timeout.
         disable_retries = ('timeout_seconds' in kwargs)
         retry_after_410 = False
+        deserialize = kwargs.pop('deserialize', True)
         while True:
             resp = func(*args, **kwargs)
             try:
                 for line in iter_resp_lines(resp):
                     # unmarshal when we are receiving events from watch,
                     # return raw string when we are streaming log
                     if watch_arg == "watch":
-                        event = self.unmarshal_event(line, return_type)
+                        if deserialize:
+                            event = self.unmarshal_event(line, return_type)
+                        else:
+                            # Only do basic JSON parsing, no deserialize
+                            event = json.loads(line)
```

Comment:
I also noticed that in this way, the resource version is not tracked like [here does](https://github.com/kubernetes-client/python/blob/d80165da21a9fec1d5abc785cb446ab16f6affdb/kubernetes/base/watch/watch.py#L119-L126).

I think we can refactor the unmarshal_event method to keep track of the resource version even when we skip deserialization.